aiohomekit/controller/ble/pairing.py

```diff
@@ -944,19 +972,12 @@ async def _async_restore_subscriptions(self) -> None:
         if not self._restore_pending or not self.client or not self.client.is_connected:
             return
 
-        logger.debug("%s: Setting broadcast encryption key", self.name)
-        async with self._ble_request_lock:
-            await self._async_set_broadcast_encryption_key()
-        long_term_pub_key_hex: str = self.pairing_data["iOSDeviceLTPK"]
-        long_term_pub_key_bytes = bytes.fromhex(long_term_pub_key_hex)
-        self._broadcast_decryption_key = BroadcastDecryptionKey(
-            self._derive(long_term_pub_key_bytes, b"Broadcast-Encryption-Key")
-        )
```

Comment:
Moved this into `_async_set_broadcast_encryption_key` so we are sure its never out of sync